### PR TITLE
Added empty option to signature

### LINKED ISSUE #626: Make the signature on Proposal optional field?
In TM: https://github.com/tendermint/tendermint/blob/master/consensus/state.go#L1089
`SignProposalRequest` is created without the signature (the raw type will have `vec![0]` as the signature), so https://github.com/informalsystems/tendermint-rs/blob/master/tendermint/src/proposal/sign_proposal.rs#L24 will fail, because the "signature" has an incorrect size.
should this be patched here to account for this case, or should Tendermint be patched (change the protobuf type or "pad" the empty signature field to the expected length)? @greg-szabo @marbar3778

### PR DESCRIPTION
**Pull request by greg-szabo**
Fixes #626 .

This is a proposed fix for optional signatures.

* [X] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [X] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: tendermint/src/signature.rs [hunk 1]
```diff
@@ -27,6 +30,9 @@ impl TryFrom<Vec<u8>> for Signature {
     type Error = Error;
 
     fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        if value.is_empty() {
```
Comment (tomtau) [inline tendermint/src/signature.rs:33]: The value sent from Go Tendermint over privval socket (for signing) in Vote/Proposal requests is actually not empty: https://github.com/informalsystems/tendermint-rs/issues/626#issue-717983039 -- it has one element which is `0u8`.
tmkms could do some preprocessing though and clear that value before creating a domain type

Comment (greg-szabo) [reply tendermint/src/signature.rs:33]: Apologies, I misunderstood your comment in the issue. And, come on, a single `0u8`? Aaaargh.

I'll add that in the code. One more thing to raise to the Go team. Thanks!

Comment (tomtau) [reply tendermint/src/signature.rs]: oops, my apologies -- the `vec![0u8]` is something I saw in a debugger when running the Go unit tests. I've just tried against the TM process (0.34.0-rc5) and it's indeed empty.
735c153faf19133305f8ab3adad9385220999a4d is OK
(9ed45f012a1d2d76b27d9d49858217f60d8281f9 can be reverted)

**Comment by tomtau** [comment]
Tested ACK -- https://github.com/informalsystems/tendermint-rs/pull/635#discussion_r504477240

File: tendermint/src/signature.rs (hunk 1 above)
Comment (greg-szabo) [reply tendermint/src/signature.rs]: LOL, thanks for checking again! I'm happy to implement extra work if it leads to extra checks! I should have checked it myself...

**Review by thanethomson — APPROVED** [review]

**Pull request merged**